Hearthstone Deck Tracker/Importing/CardImageImporter.cs

```diff
@@ -0,0 +1,108 @@
+﻿using Hearthstone_Deck_Tracker.Hearthstone.Entities;
+using Hearthstone_Deck_Tracker.Utility.Logging;
+using System;
+using System.Collections.Generic;
+using System.IO;
+using System.Linq;
+using System.Net;
+using System.Text;
+using System.Threading.Tasks;
+
+namespace Hearthstone_Deck_Tracker.Importing
+{
+	public static class CardImageImporter
+	{
+		const string HearthstoneArtUrl = "https://art.hearthstonejson.com/v1/256x";
+		private static string StoreImagesPath = Path.Combine(Config.AppDataPath, "\\Images");
+
+		private static List<string> _succesfullyDownloadedImages = new List<string>();
+		private static Dictionary<string, Task> _inProcessDownloads = new Dictionary<string, Task>();
+
+		static CardImageImporter()
+		{
+			if(!Directory.Exists(StoreImagesPath))
+				Directory.CreateDirectory(StoreImagesPath);
+			_succesfullyDownloadedImages.AddRange(GetCurrentlyStoredCardids());
+		}
+
+		public static async Task FinishDownloadingEntites(Entity[] entities)
+		{
+			var toAwait = new List<Task>();
+			foreach(var entity in entities)
+			{
+				if(CardImageIsDownloaded(entity))
+				{
+					continue;
+				}
+				if(_inProcessDownloads.TryGetValue(entity.CardId, out var task))
+				{
+					if(task.IsCompleted)
```

Comment:
You may want to differentiate between `IsCompelted` and `IsCompletedSuccessfully` here. From the `IsCompleted` documentation:
> true if the task has completed (that is, the task is in one of the three final states: RanToCompletion, Faulted, or Canceled); otherwise, false.
https://docs.microsoft.com/en-us/dotnet/api/system.threading.tasks.task.iscompleted?view=netcore-3.1